segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -596,19 +601,21 @@ public void validateAndSaveSnapshot(MetadataTransaction txn) throws Exception {
      * @param bytes Bytes to write.
      */
     private void writeToJournal(ByteArraySegment bytes) throws ExecutionException, InterruptedException {
-        val bytesWritten = chunkStorage.write(currentHandle, systemJournalOffset, bytes.getLength(),
-                new ByteArrayInputStream(bytes.array(), bytes.arrayOffset(), bytes.getLength())).get();
-        Preconditions.checkState(bytesWritten == bytes.getLength());
-        systemJournalOffset += bytesWritten;
-    }
 
-    /**
-     * Adds a new System journal file.
-     */
-    private void startNewJournalFile() throws ExecutionException, InterruptedException {
-        currentHandle = chunkStorage.create(getSystemJournalChunkName(containerId, epoch, currentFileIndex + 1)).get();
-        currentFileIndex++;
-        systemJournalOffset = 0;
+        if (newChunkRequired) {
+            currentFileIndex++;
+            systemJournalOffset = 0;
+            currentHandle = chunkStorage.createWithContent(getSystemJournalChunkName(containerId, epoch, currentFileIndex), bytes.getLength(),
```

Comment:
Are you holding any locks while modifying all this state?

If so, then the same comment as above regarding holding locks while performing async tasks.

Reply:
same as above.